modules/java/native/java-grammar.ym

```diff
@@ -91,6 +94,9 @@ java_dest_custom_options
 java_dest_custom_option
   : string string_or_number { java_dd_set_option(last_driver, $1, $2); free($1); free($2); }
 
+java_global_option
+        : KW_JVM_OPTIONS '(' string ')' {java_config_set_jvm_options(configuration, g_strdup($3)); free($3);}
```

Comment:
Specifying the jvm_options more than once causes a small memory leak (this is a very rare case, but our grammar allows it).
You can call `g_strdup()` in `java_config_set_jvm_options()`. The previous instance can be freed in that method as well.

We have a setter pattern for this, for example:
https://github.com/balabit/syslog-ng/blob/9ffca9f8b4d3402ce2a960db492d2704c4d843e0/modules/xml/xml.c#L269-L277